racket/collects/net/url-string.rkt

```diff
@@ -176,17 +176,20 @@
            "(?:"              ; | / user-at-opt
            "([^/?#@]*)"       ; | | #2 = user-opt
            "@)?"              ; | \
-           "([^/?#:]*)?"      ; | #3 = host-opt
+           "(\\["             ; | / #3 = ipv6-host-opt
+           "(?:.*)"           ; | | hex-addresses
+           "\\])?"            ; | \
+           "([^/?#:]*)?"      ; | #4 = host-opt
```

Comment:
This reads like `(ipv6-host-regexp)?(host-regexp)?`, so allows both to be present and, below, discards the host using the IPv6 host instead. Don't we rather want to use alternation, as in `((ipv6-regexp)|(host-regexp))?` (allowing either one, or nothing since the host is optional, but not both)?

The [PR with tests](https://github.com/racket/net/pull/8/files) lacks negative tests against this.